service/auth.go

```diff
@@ -0,0 +1,67 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	authclient "github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+type Auth interface {
+	CheckSpaceScope(ctx context.Context, spaceID, requiredScope string) (bool, error)
+}
+
+func NewAuthService(hostURL string) (Auth, error) {
+	u, err := url.Parse(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client := http.Client{}
+	c := authclient.New(&doer{
+		target: goaclient.HTTPClientDoer(&client),
+	})
+	c.Host = u.Host
+	c.Scheme = u.Scheme
+	return &auth{c}, nil
+}
+
+type doer struct {
+	target goaclient.Doer
+}
+
+func (d *doer) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	token := jwt.ContextJWT(ctx)
+	if token != nil {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Raw))
+	}
+	return d.target.Do(ctx, req)
+}
+
+type auth struct {
+	*authclient.Client
+}
+
+func (a *auth) CheckSpaceScope(ctx context.Context, spaceID, requiredScope string) (bool, error) {
```

Comment:
I think we should have two methods here.
One returns (bool, error) and the other just error,
If the user doesn't have required scope then we return errors.NewForbiddenError() (which will converted to 403 by client) error. I think it will simplify usage in many clients when we could do just:
```
err := auth.RequireScope(ctx, spaceID, scope)
if err != nil {
    return err
    // Or return jsonapi.JSONErrorResponse(ctx, err) if in controller.
}
// User has permissions. Go ahead and do some work here.
...
```
Similar to what we have in Auth - https://github.com/fabric8-services/fabric8-auth/blob/master/authorization/permission/service/permission_service.go#L31-L54

Reply:
@alexeykazakov I'm not sure why you want to have 2 functions. I believe it's going to be confusing for the user. Could you elaborate/clarify on the purpose of the 2 functions you had in mind, please?

Reply:
For now, I will only keep one method as below.
`CheckResourceScope(ctx context.Context, resourceID, requiredScope string) error`

Reply:
done [e1b3d2a](https://github.com/fabric8-services/fabric8-common/pull/62/commits/e1b3d2a5f9b084532eec637dc6fac7546edde1f9)